Refactor episode 4 and 5 (#154)
ESMValGroup/ESMValTool_Tutorial: _episodes/04-recipe.md

> >        `select_january` to the variable `tas`.
> >      - a diagnostic task called `script1`, applying the script
> >        `examples/diagnostic.py` to the preprocessed data (`map/tas`).
> >
> >      Under the diagnostic `timeseries` we find three tasks:
> >      - a preprocessor task called `tas_amsterdam`, applying the preprocessor
> >        called `annual_mean_amsterdam` to the variable `tas`.
> >      - a preprocessor task called `tas_global`, applying the preprocessor
> >        called `annual_mean_global` to the variable `tas`.
> >      - a diagnostic task called `script1`, applying the script
> >        `examples/diagnostic.py` to the preprocessed data
> >        (timeseries/tas_global and timeseries/tas_amsterdam).
> >
> {: .solution}
{: .challenge}

> ## Pro tip: short names and variable groups
>
> The preprocessor tasks in ESMValTool are called 'variable groups'. For the
> diagnostic `timeseries`, we have two variable groups: `tas_amsterdam` and
> `tas_global`. Both of them operate on the variable `tas` (as indicated by the
> `short_name`), but they apply different preprocessors. For the diagnostic

> > `map` the variable group itself is named `tas`, and you'll notice that we're
> > not explicitly providing the `short_name`. This is a shorthand build into

Reviewer: ```suggestion
> `map` the variable group itself is named `tas`, and you'll notice that we do
> not explicitly provide the `short_name`. This is a shorthand build into
```

> ESMValTool.
{: .callout}

> ## Output files
>

> > Have another look at the output directory created by the ESMValTool run.

Reviewer: Should this point to the work directory? Because my output directory does not have the directories listed below.

Author: I think in the config episode we are setting the output_dir, right? 
Otherwise, we can use the log output of ESMValTool to locate it.

>
> Which files/folders are created by each task?
>
> > ## Answer
> >
> > - **map/tas**: creates `/preproc/map/tas`, which contains preprocessed data
> >   for each of the input datasets, and a file called `metadata.yml`
> >   describing the contents of these datasets.
> > - **timeseries/tas_global**: creates `/preproc/timeseries/tas_global`, which
> >   contains preprocessed data for each of the input datasets, and
> >   `metadata.yml`.
> > - **timeseries/tas_amsterdam**: creates `/preproc/timeseries/tas_amsterdam`,
> >   which contains preprocessed data for each of the input datasets, plus a

> > >   combined 'MultiModelMean', and `metadata.yml`.

Reviewer: ```suggestion
> >   combined `MultiModelMean`, and `metadata.yml`.
```

> >   respectively. For each output file, there is also corresponding provenance
> >   information in the form of `.svg`, `.xml`, `.bibtex` and `.txt` files.
> > - **timeseries/script1**: creates `/run/timeseries/script1` with general
> >   information and a log of the diagnostic script run. It also creates
> >   `/plots/timeseries/script1` and `/work/timeseries/script1`, which contain
> >   output figures and output datasets, respectively. For each output file,
> >   there is also corresponding provenance information in the form of `.svg`,
> >   `.xml`, `.bibtex` and `.txt` files.
> >
> {: .solution}
{: .challenge}

> ## Pro tip: diagnostic logs
>
> When you run ESMValTool, any log messages from the diagnostic script are not printed on the terminal.
> But they are written to the `log.txt` files in the folder `/run/<diag_name>/log.txt`.
>
> ESMValTool *does* print a command that can be used to re-run a diagnostic script. When you use this
> the output *will* be printed to the command line.
{: .callout}

## Modifying the example recipe

Let's make a small modification to the example recipe.

> ## Change your location
>

> > Modify the recipe to analyse the temperature for your own location.

Reviewer: ```suggestion
> Modify and run the recipe to analyse the temperature for your own location.
```

> > Modify the recipe to analyse the temperature for your own location.

Reviewer: I suggest to add a note here to run the local file, i.e. `esmvaltool run example_python.yml` so that they do not use the previous command and wonder why it does not work.

>
> > ## Solution
> >
> > In principle, you only have to modify the latitude and longitude coordinates

> > > in the preprocessor called 'annual_mean_amsterdam'. However, it is good

Reviewer: ```suggestion
> > in the preprocessor called `annual_mean_amsterdam`. However, it is good
```

> > > practice to also replace all instances of 'amsterdam' with the correct name

Reviewer: ```suggestion
> > practice to also replace all instances of `amsterdam` with the correct name
```

> > of your location. Otherwise the log messages and output will be confusing.
> > You are free to modify the names of preprocessors or diagnostics.
> >
> > In the `diff` file below you will see the changes we have made to the file.

> > > The top 2 lines are the filenames and the lines like @@ -29,10 +29,10 @@

Reviewer: ```suggestion
> > The top 2 lines are the filenames and the lines like `@@ -29,10 +29,10 @@`
```
